CRVResponse/src/MakeCrvDigis.cc

```diff
@@ -8,13 +8,12 @@ void MakeCrvDigis::SetWaveform(const std::vector<double> &waveform, double ADCco
   _ADCs.clear();
   for(size_t i=0; i<waveform.size(); i++)
   {
-    if(waveform[i]*ADCconversionFactor+pedestal>0) _ADCs.push_back(static_cast<unsigned int>(waveform[i]*ADCconversionFactor+pedestal+0.5));
-    else _ADCs.push_back(0);
+    _ADCs.push_back(static_cast<int16_t>(waveform[i]*ADCconversionFactor+pedestal+0.5));
   }
 
   int TDCtmp=lrint(startTime/digitizationPrecision);
   if(TDCtmp<0) throw std::logic_error("ERROR: found a waveform start time (relative to the event marker) < 0"); //this shouldn't happen
-  _TDC=static_cast<unsigned int>(TDCtmp);
+  _TDC=static_cast<uint16_t>(TDCtmp);
```

Comment:
This will truncate in case TDCtmp > 65535.  Is that what the hardware does?  Or does it saturate?  If it saturates, you should add an std::max(TDCtmp,std::numeric_limits<uint16_t>.max()));

Reply:
the current implementation for the timestamp has only 12 bit (--> 2^12*12.55ns=0.05ms) which is insufficient for the interspill microbunch. i think the hardware starts over at 0 for timestamps larger than 2^12.
one proposal is to change it to 16 bit (--> 2^16*12.55ns=0.8ms).

Reply:
Just checking: I believe that the code you are committing now will support the 100 micro-second Offspill event windows.  Is that right?

Reply:
Yes, the uint16_t supports the 100us OffSpill windows. However, the firmware still requires some changes for that (currently it only uses 12 bits / up to 50us). Once this is done, we will need another change in the CRVFragment Overlay class in mu2e_artdaq_core.